src/psyclone/psyir/symbols/symbol.py

```diff
@@ -255,6 +255,32 @@ def copy(self):
         return type(self)(self.name, visibility=self.visibility,
                           interface=self.interface)
 
+    def specialise(self, subclass):
+        '''Specialise this class instance so that it becomes an instance of
```

Comment:
Suggestions to improve readability (subjective, I won't mind if not taken):
s/class instance/symbol/
s/references to this instance/references to it/

Also show intent to the param, e.g.: the new subclass that will describe this symbol.
and type: \`type\` of a Symbol subclass

Reply:
Updated. You probably won't like my definition of the type of subclass. The issue is that the type of all classes is `type` so I feel the need to point this out.